sql/src/test/java/org/apache/druid/sql/calcite/BaseCalciteQueryTest.java

```diff
@@ -295,26 +280,13 @@ public static Map<String, Object> getTimeseriesContextWithFloorTime(
 
   public static final Map<String, Object> OUTER_LIMIT_CONTEXT = new HashMap<>(QUERY_CONTEXT_DEFAULT);
 
-  @Nullable
-  public final SqlEngine engine0;
   final boolean useDefault = NullHandling.replaceWithDefault();
 
   public boolean cannotVectorize = false;
   public boolean skipVectorize = false;
 
-  private QueryComponentSupplier baseComponentSupplier;
   public PlannerComponentSupplier basePlannerComponentSupplier = new StandardPlannerComponentSupplier();
```

Comment:
there was a separate PR to [change this](https://github.com/apache/druid/pull/16245); however I've came to the conclusion that whatever this `PlannerComponentSupplier` wanted to be - it never really started getting used.

its not used in the current `master` ; providing the default implementation like this is enough.

but...if it needs to be used somewhere - maybe the `QueryComponentSupplier` could just have a new interface method which returns the `PlannerComponentSupplier` stuff

Reply:
made the following changes:
* added a `QueryComponentSupplier#getPlannerComponentSupplier` method
* `StandardComponentSupplier#buildPlannerComponentSupplier` to enable implementation to only build it once `StandardComponentSupplier`
* `SqlTestFramework#plannerFixture` doesn't anymore need `PlannerComponentSupplier` to be passed separetly (as its defined implicitly by the `QueryComponentSupplier`

I think this may make it possible to retain control over these things - custom implementation could move their custom `PlannerComponentSupplier` into their custom `QueryComponentSupplier`